libsignal-service/src/lib.rs

```diff
@@ -9,3 +9,36 @@ pub const USER_AGENT: &'static str =
     concat!(env!("CARGO_PKG_NAME"), "-rs-", env!("CARGO_PKG_VERSION"));
 
 pub struct TrustStore;
+
+/// Attachment represents an attachment received from a peer
+///
+/// Source: `textsecure/textsecure.go`
```

Comment:
Commented a bit on IRC as well, but referring to the Java sources is a frustrating timesink... https://github.com/signalapp/libsignal-service-java/blob/master/java/src/main/java/org/whispersystems/signalservice/api/messages/SignalServiceAttachment.java#L105 might be the file i/o part (and naturally the file is the class), but this is completely different from Go.

I think it might be a good idea to either drop the Go references or spend a short eternity on making this code closer to Java and then beating Whisperfish into working with that.

To get things done, I'd rather drop the comments and after Whisperfish is able to receive a message, look more into how it's done in Java.

Reply:
Ok. Keep it as close as possible to Java, I'd say, but don't go through FactoryBuilderFactory hoops, and there's no need to make all the logic already, since it will be a lot cleaner in Rust on first sight.

Reply:
Well, let's see what turns out to be feasible, because the Java stuff is so crazy different... Or at least it has more stuff which we don't even know how to store yet, but maybe the fields could go in soon-ish and then be ignored in Whisperfish.

Reply:
And here we discussed to go with the Go approach at least for now. The Java version seems to dump encrypted bytes and leave decrypting to the client, whereas Whisperfish does it differently.

We can always revisit this in the future and add fields that aren't there currently through Go.

Reply:
And I left the Go comments in because even if no one knows about the Go version, now they do ;) Will remove if asked to.